Console/MWD.cpp

```diff
@@ -0,0 +1,678 @@
+
+    return TRUE;
+}
+
+bool MWDComm::Run()		// wywo�ywanie obs�ugi MWD + generacja wi�kszego op�nienia
+{
+    MWDTime++;
+    if(!(MWDTime%5))
+    {
+        if(GetMWDState())
+        {
+            SendData();
+            ReadData();
+            return 1;
+        }else
+        {
+            WriteLog("Port COM: blad polaczenia!");
+            // mo�e spr�bowa� si� po��czy� znowu?
+            return 0;
+        }
+        MWDTime = 0;
+    }
+}
+
+bool MWDComm::CheckData()	// sprawdzanie wej�� cyfrowych i odpowiednie sterowanie maszyn�
+{
+    int i=0;
+    while(i<6)
+    {
+        maskData[i] = ReadDataBuff[i]^lastStateData[i];
+        lastStateData[i] = ReadDataBuff[i];
+        i++;
+    }
+    /*
+                Rozpiska port�w!
+                Port0: 	0 	NC			odblok. przek. spr�arki i wentyl. opor�w
+                                1 	M			wy��cznik wy�. szybkiego
+                                2 	Shift+M		impuls za��czaj�cy wy�. szybki
+                                3 	N			odblok. przeka�nik�w nadmiarowych i r�nicowego obwodu g��wnego
+                                4 	NC			rezerwa
+                                5 	Ctrl+N		odblok. przek. nadmiarowych przetwornicy, ogrzewania poci�gu i r�nicowych obw. pomocniczych
+                                6 	L			wy�. stycznik�w liniowych
+                                7 	SPACE		kasowanie czuwaka
+
+                Port1: 	0 	NC
+                                1 	(Shift)	X	przetwornica
+                                2 	(Shift)	C	spr�arka
+                                3 	S			piasecznice
+                                4 	(Shift)	H	ogrzewanie sk�adu
+                                5 				przel. hamowania	Shift+B pspbpwy Ctrl+B pospieszny B towarowy
+                                6 				przel. hamowania
+                                7 	(Shift)	F	rozruch w/n
+
+                Port2: 	0 	(Shift) P	pantograf przedni
+                                1 	(Shift)	O	pantograf tylni
+                                2 	ENTER		przyhamowanie przy po�lizgu
+                                3 	()		przyciemnienie �wiate�
+                                4 	()		przyciemnienie �wiate�
+                                5 	NUM6		odlu�niacz
+                                6 	a			syrena lok W
+                                7	A			syrena lok N
+
+                Port3: 	0 	Shift+J		bateria
+                                1
+                                2
+                                3
+                                4
+                                5
+                                6
+                                7
+        */
+
+    /*	po prze��czeniu bistabilnego najpierw wciskamy klawisz i przy nast�pnym
+                wej�ciu w p�tl� MWD puszczamy bo inaczej nie dzia�a							*/
+
+    // wciskanie przycisk�w klawiatury
+    /*PORT0*/
+    if(maskData[0] & 0x02) if(lastStateData[0] & 0x02) 	KeyBoard('M',1);	// wy��czenie wy��cznika szybkiego

```

Comment:
Czy to ma działać tak, że włączamy po drugim przeczytaniu informacji a wyłączamy po pierwszym przeczytaniu? To nie reaguje na zmiany stanu. Tyczy się wszystkich warunków niżej.
I nie widzę sensu w robieniu dwóch if-ów jeśli można zrobić jednym.